specs/menu/menu.md

```diff
@@ -0,0 +1,210 @@
+            <fast-menu-item>Nested item two</fast-menu-item>
+            <fast-menu-item>
+                Nested item three
+                <fast-menu slot="item">
+                    <fast-menu-item>Nested item one</fast-menu-item>
+                    <fast-menu-item>Nested item two</fast-menu-item>
+                </fast-menu>
+            </fast-menu-item>
+        </fast-menu>
+    </fast-menu-item>
+    <fast-menu-item>Menu item four</fast-menu-item>
+</fast-context-menu>
+```
+
+#### Icon support:
+
+Consider a scenario where we have the following menu - some menu items have glyphs, others do not:
+```html
+<fast-menu>
+    <fast-menu-item>Menu item one</fast-menu-item>
+    <fast-menu-item>Menu item two</fast-menu-item>
+    <fast-menu-item>
+        <fancy-icon slot="icon"></fany-icon>
+        Menu item three
+    </fast-menu-item>
+</fast-context-menu>
+```
+The third item in the menu has a glyph, which would inset the text from the other menu items. To ensure that the content of this menu item aligns with the rest of the items, all items need to visually shift to ensure alignmnet of the menu content. This presents an interesting challenge for how we will communicate these changes to ensure the styling is appropriately applied. Ultimately, the parent just needs to know that *one* of the items below has a menu item with the glyph.
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/context-menu)
+- [Material UI](https://material-ui.com/components/menus/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/menus/)
+- [Ant Design - menu](https://ant.design/components/menu/)
+- [Ant Design - dropdown](https://ant.design/components/menu/)
+- [Atlassian - menu](https://atlaskit.atlassian.com/packages/core/menu)
+- [Atlassian - dropdown menu](https://atlaskit.atlassian.com/packages/core/dropdown-menu)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/tree-view)
+
+---
+
+## Design
+
+### API
+
+*The key elements of the component's public API surface:*
+
+**Menu**
+*Component name:*
+- `fast-menu`
+
+*Attributes:*
+- extends HTML Element attributes
+
+*Slots:*
+- default slot for items
+
+*Events*
+- none
+
+**Menu Item**
+*Component name:*
+- `fast-menu-item`
+
+*Attributes:*
+- role - an enum representing the menu items' role
+    - menuitem
+    - menuitemcheckbox
+    - menuitemradio
+- disabled - the menu item is disabled
+- checked - sets the checked value for menuitemcheckbox or menuitemradio items
+
+*Slots:*
+- content - slot for the content (the default slot for the item)
+- item - the optional children
+- before-content - slot which precedes content, often a glyph or icon
+- after-content - slot which follows content, often a glyph or icon
+
+*CSS Parts*
+- content
+- before-content
+- after-content
+- item
+
+*Events:*
+- invoke (event) - event for when the item has been clicked or invoked via keyboard
```

Comment:
a `<button>` element fires a "click" event on mouse click or Enter. Did you consider mapping to that behavior?

Reply:
As in mapping the behavior of a menuitem to a click event? We could, though the behavior of what happens on enter/space are slightly different depending on the role.

Reply:
Yea reusing the "click" event name. Checkbox and radio inputs both *also* fire a click event on space / mouse click (as well as an input and change event). My thought is that since there is already precedent for the behavior we desire from native elements that we should align rather than making up new event name.

Reply:
The difference in behavior is spelled out when we reference the W3C keyboard interaction. I don't think we need different events though to manage it. The primary differences depend on the role of the item and state of the menu and any submenus. It seems reasonable to reuse the click event for keyboard here.